cmd/admin-heal-status.go

```diff
@@ -0,0 +1,320 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"golang.org/x/tools/container/intsets"
+
+	humanize "github.com/dustin/go-humanize"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/minio/pkg/madmin"
+)
+
+var adminHealStatusCmd = cli.Command{
+	Name:            "status",
+	Usage:           "Status of a heal task",
+	Before:          setGlobalsFromContext,
+	Action:          mainAdminHealStatus,
+	Flags:           globalFlags,
+	HideHelpCommand: true,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+    1. To fetch status of a running heal task with no bucket and object-prefix in a Minio server with alias 'play'
+       $ {{.HelpName}} play
+
+    2. To fetch status of a running heal task on bucket 'testbucket' in a Minio server with alias 'play'
+       $ {{.HelpName}} play/testbucket/
+
+`,
+}
+
+// wrapRowsAndPrint - given content for a single row in a table, this
+// function wraps and prints the content for each cell of the table to
+// use multiple lines, while respecting the maximum column width.
+func wrapRowsAndPrint(t *console.Table, col []string) {
+	w := t.ColWidth
+	todo := true
+	for l := 0; todo; l++ {
+		pcol := make([]string, len(col))
+		isNonEmpty := false
+		for i, v := range col {
+			if len(v) <= l*w[i] {
+				continue
+			}
+			// data remains to be written
+			if len(v)-l*w[i] > w[i] {
+				pcol[i] = v[l*w[i] : (l+1)*w[i]]
+			} else {
+				pcol[i] = v[l*w[i]:]
+			}
+			isNonEmpty = true
+		}
+		// fmt.Println(pcol)
+		if isNonEmpty {
+			t.PrintRow(pcol)
+		} else {
+			todo = false
+		}
+	}
+}
+
+func makeBarChart(done, total int64) string {
+	length := 15
+	p := int(done * int64(length) / total)
+	return fmt.Sprintf("[%s%s]", strings.Repeat("#", p),
+		strings.Repeat(".", length-p))
+}
+
+func makeHealEntityString(h madmin.HealResultItem) string {
+	switch h.Type {
+	case madmin.HealItemObject:
+		return h.Bucket + "/" + h.Object
+	case madmin.HealItemBucket:
+		return h.Bucket
+	case madmin.HealItemMetadata:
+		return "[disk-format]"
+	case madmin.HealItemBucketMetadata:
+		return fmt.Sprintf("[bucket-metadata]%s/%s", h.Bucket, h.Object)
+	case madmin.HealItemMultipartUpload:
+		return fmt.Sprintf("[multipart-upload]%s/%s(%s)", h.Bucket,
+			h.Object, h.Detail)
+	}
+	return "(unknown - this is a bug)"
```

Comment:
we could add a line asking them to open an issue in github.com/minio/mc/issues.

Reply:
errUnexpected can be used.